### ISSUE TITLE
loss goes to NaN after 600K steps

### ISSUE THREAD

**Issue by sun-peach** [SWivid/F5-TTS#790]
### Checks

- [x] This template is only for bug reports, usage problems go with 'Help Wanted'.
- [x] I have thoroughly reviewed the project documentation but couldn't find information to solve my problem.
- [x] I have searched for existing issues, including closed ones, and couldn't find a solution.
- [x] I confirm that I am using English to submit this report in order to facilitate communication.

### Environment Details

Use Ubuntu, torch 2.3.1+cu121, CUDA 12.1

### Steps to Reproduce

runing on 8 H100 cards with the command: `CUDA_VISIBLE_DEVICES=0,1,2,3,4,5,6,7 accelerate launch --num_processes 8 --mixed_precision fp16 --multi_gpu --main_process_port 25901 train.py --config-name F5TTS_Base_train.yaml`

The F5TTS_Base_train.yaml file contains the config listed below:

> hydra:
>   run:
>     dir: ckpts/${model.name}_${model.mel_spec.mel_spec_type}_${model.tokenizer}_${datasets.name}/${now:%Y-%m-%d}/${now:%H-%M-%S}
>     
> datasets:
>   name: Emilia_ZH_EN  # dataset name
>   batch_size_per_gpu: 38400  # 8 GPUs, 8 * 38400 = 307200
>   batch_size_type: frame  # "frame" or "sample"
>   max_samples: 64  # max sequences per batch if use frame-wise batch_size. we set 32 for small models, 64 for base models
>   num_workers: 128
> 
> optim:
>   epochs: 15
>   learning_rate: 7.5e-5
>   num_warmup_updates: 20000  # warmup updates
>   grad_accumulation_steps: 1  # note: updates = steps / grad_accumulation_steps
>   max_grad_norm: 1.0  # gradient clipping
>   bnb_optimizer: False  # use bnb 8bit AdamW optimizer or not
> 
> model:
>   name: F5TTS_Base_norm_flow_8GPU  # model name
>   tokenizer: pinyin  # tokenizer type
>   tokenizer_path: None  # if tokenizer = 'custom', define the path to the tokenizer you want to use (should be vocab.txt)
>   arch:
>     dim: 1024
>     depth: 22
>     heads: 16
>     ff_mult: 2
>     text_dim: 512
>     conv_layers: 4
>     checkpoint_activations: False  # recompute activations and save memory for extra compute
>   mel_spec:
>     target_sample_rate: 24000
>     n_mel_channels: 100
>     hop_length: 256
>     win_length: 1024
>     n_fft: 1024
>     mel_spec_type: vocos  # 'vocos' or 'bigvgan'
>   vocoder:
>     is_local: False  # use local offline ckpt or not
>     local_path: None  # local vocoder path
> 
> ckpts:
>   logger: wandb  # wandb | tensorboard | None
>   save_per_updates: 10000  # save checkpoint per steps
>   last_per_steps: 5000  # save last checkpoint per steps
>   save_dir: /data4/F5TTS/ckpts/${model.name}_${model.mel_spec.mel_spec_type}_${model.tokenizer}_${datasets.name}

### ✔️ Expected Behavior

Go down to 15 epochs to finish the experiment.

### ❌ Actual Behavior

After about 600k steps, its loss turns to NaN

**Comment by ZhikangNiu** [comment]
Have you changed the arch of F5? So what is the arch of F5TTS_Base_norm_flow_8GPU

**Comment by sun-peach** [comment]
No, I use the original model CFM:

`model = CFM(
        transformer=model_cls(**cfg.model.arch, text_num_embeds=vocab_size, mel_dim=cfg.model.mel_spec.n_mel_channels),
        mel_spec_kwargs=cfg.model.mel_spec,
        vocab_char_map=vocab_char_map,
    )`

**Comment by SynthAether** [comment]
I experienced a similar NaN loss after 700K steps using 8 GPUs and settings very similar to yours. I am using an internal multi-speaker dataset comprising approximately 750 hours of clean speech. I suspected that the `-mixed_precision fp16` flag might have been causing the issue, so I resumed training from an earlier checkpoint with full `fp32` precision. This required me to reduce the batch size to avoid out-of-memory errors. While this allowed training without encountering any NaN errors, and the training went fine to two million steps, I noticed that the quality of the TTS output dropped significantly (e.g. word skipping, mispronounced words, etc.). I don't know if this means the batch size is important or that switching from `fp16` to `fp32` mid-training has adverse effects.

**Comment by ZhikangNiu** [comment]
> I experienced a similar NaN loss after 700K steps using 8 GPUs and settings very similar to yours. I am using an internal multi-speaker dataset comprising approximately 750 hours of clean speech. I suspected that the `-mixed_precision fp16` flag might have been causing the issue, so I resumed training from an earlier checkpoint with full `fp32` precision. This required me to reduce the batch size to avoid out-of-memory errors. While this allowed training without encountering any NaN errors, and the training went fine to two million steps, I noticed that the quality of the TTS output dropped significantly (e.g. word skipping, mispronounced words, etc.). I don't know if this means the batch size is important or that switching from `fp16` to `fp32` mid-training has adverse effects.

Maybe you need to check this one
https://github.com/SWivid/F5-TTS/blob/a72f2f8efb72b90a2827d2f02cae91b070790dd2/src/f5_tts/eval/eval_infer_batch.py#L157

**Comment by SWivid** [comment]
will have discussion along #832
